packages/client/src/components/preview/DNDPlaceholderOverlay.svelte

```diff
@@ -0,0 +1,44 @@
+<script>
+  import { onMount } from "svelte"
+  import { DNDPlaceholderID } from "constants"
+
+  let left, top, height, width
+
+  onMount(() => {
+    const interval = setInterval(() => {
```

Comment:
Appreciate the detailed review!

Quick checklist of changes I've made:
- [x] Add handlers to blocks (noticed this myself, but forgot to ever add it)
- [x] Use `requestAnimationFrame` for DNDPlaceholderOverlay
- [x] Fix layout shifting on drop
- [x] Hide new component panel when selected component changes 

> Could be wrong, but this appears to be a good use case for requestAnimationFrame. Although I ran the chrome profiler against your branch and there doesn't seem to be any problems with layout thrashing/repaints so it's probably fine +1

Good call - we do this already for the other overlays in the client previews (like the selection indicators) so I've added it in here too. Since we only update these once every 100ms it doesn't actually affect performance by using `requestAnimationFrame`, but the big gain is actually on battery life, since wrapping functions inside `requestAnimationFrame` will stop them running when the window loses focus or the tab is changed.

> Just another one in case you were interested - Chrome does however pick up that Layout shifts are occurring when the elements are getting dropped - because they kind of are (when the elements contract back after dropping your element) But it's no big deal in this case. Not sure how you would actually solve for that.

I tried an idea here which I think works pretty well - if we only add the `transition` CSS rule to components that are expanded in this way, then they'll animate initially but shrink instantly when dropping happens, since they'll lose the transition rule at the same moment. That feels like a big reduction in the layout shifting but I can't actually test it because Chrome has never worked with the dev env for me haha.

> Once an element is dragged onto the screen, you must exit the component menu manually to see the settings for that component. I kind of expected to see the settings straight away, but can understand that you may want to drag a few onto the screen at a time. However - when I click the component inside the preview, the component menu remains (which is the default behaviour even before this) but I think the component menu should be hidden when you click on an element in the preview. When you click the element in the preview, you don't get any feedback, so it feels like something isn't working - when that's not the case.

Yea I agree that this felt weird. Like you said - it's just a side effect of the current behaviour, so I think it's time to change that. Changing the selected component will now close the new component panel, which I definitely think feels better.